lib/packwerk/file_processor.rb

```diff
@@ -16,24 +18,43 @@ def initialize(node_processor_factory:, parser_factory: nil)
       @parser_factory = parser_factory || Packwerk::Parsers::Factory.instance
     end
 
+    sig do
+      params(file_path: String).returns(
+        T::Array[
+          T.any(
+            Packwerk::Reference,
+            Packwerk::Offense,
```

Comment:
`FileProcessor` still have chance to early-return Offence, in cases of `UnknownFileTypeResult` or `ParseError`.

I don't particular like this shared responsibility of the class, and more than happy to revisit the further separation.